chirp/drivers/baofeng_uv17Pro.py

```diff
@@ -0,0 +1,1427 @@
+        mem.extra.append(rs)
+
+        if (_mem.pttid != 0xFF):
+            rs = RadioSetting("pttid", "PTT ID",
+                              RadioSettingValueList(self.PTTID_LIST,
+                                                    self.PTTID_LIST[
+                                                        _mem.pttid]))
+            mem.extra.append(rs)
+
+        if (_mem.scode != 0xFF):
+            rs = RadioSetting("scode", "S-CODE",
+                              RadioSettingValueList(self.SCODE_LIST,
+                                                    self.SCODE_LIST[
+                                                        _mem.scode]))
+            mem.extra.append(rs)
+
+        return mem
+
+    def set_memory(self, mem):
+        _mem = self._memobj.memory[mem.number]
+
+        _mem.set_raw("\x00"*16 + "\xff" * 16)
+
+        if mem.empty:
+            _mem.set_raw("\xff" * 32)
+            return
+
+        _mem.rxfreq = mem.freq / 10
+
+        if mem.duplex == "off":
+            for i in range(0, 4):
+                _mem.txfreq[i].set_raw("\xFF")
+        elif mem.duplex == "split":
+            _mem.txfreq = mem.offset / 10
+        elif mem.duplex == "+":
+            _mem.txfreq = (mem.freq + mem.offset) / 10
+        elif mem.duplex == "-":
+            _mem.txfreq = (mem.freq - mem.offset) / 10
+        else:
+            _mem.txfreq = mem.freq / 10
+
+        _namelength = self.get_features().valid_name_length
+        for i in range(_namelength):
+            try:
+                _mem.name[i] = mem.name[i]
+            except IndexError:
+                _mem.name[i] = "\xFF"
+
+        rxmode = txmode = ""
+        if mem.tmode == "Tone":
+            _mem.txtone = int(mem.rtone * 10)
+            _mem.rxtone = 0
+        elif mem.tmode == "TSQL":
+            _mem.txtone = int(mem.ctone * 10)
+            _mem.rxtone = int(mem.ctone * 10)
+        elif mem.tmode == "DTCS":
+            rxmode = txmode = "DTCS"
+            _mem.txtone = self.DTCS_CODES.index(mem.dtcs) + 1
+            _mem.rxtone = self.DTCS_CODES.index(mem.dtcs) + 1
+        elif mem.tmode == "Cross":
+            txmode, rxmode = mem.cross_mode.split("->", 1)
+            if txmode == "Tone":
+                _mem.txtone = int(mem.rtone * 10)
+            elif txmode == "DTCS":
+                _mem.txtone = self.DTCS_CODES.index(mem.dtcs) + 1
+            else:
+                _mem.txtone = 0
+            if rxmode == "Tone":
+                _mem.rxtone = int(mem.ctone * 10)
+            elif rxmode == "DTCS":
+                _mem.rxtone = self.DTCS_CODES.index(mem.rx_dtcs) + 1
+            else:
+                _mem.rxtone = 0
+        else:
+            _mem.rxtone = 0
+            _mem.txtone = 0
+
+        if txmode == "DTCS" and mem.dtcs_polarity[0] == "R":
+            _mem.txtone += 0x69
+        if rxmode == "DTCS" and mem.dtcs_polarity[1] == "R":
+            _mem.rxtone += 0x69
```

Comment:
Same here, `split_tone_decode()`

Reply:
I assume you mean "split_tone_encode". Fixed it. :-)